### ISSUE TITLE
Design question: Should ClusterClass & Topology be separated in the azure-aks-aso chart?

### ISSUE THREAD

**Issue by rbjorklin** [mboersma/cluster-api-charts#34]
Hello @mboersma!

First of all thanks for putting these charts together! Now over to my question: looking at the [aks-aso-clusterclass](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template-aks-aso-clusterclass.yaml) and [aks-aso-topology](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template-aks-aso-topology.yaml) these are separate flavours, which I believe is on purpose, as it allows creating a single ClusterClass and then stamping out several clusters from that one class. This does not look like it is possible with the `azure-aks-aso` chart in it's current form, is this intentional?

**Comment by mboersma** [comment]
@rbjorklin you're right that the `azure-aks-aso` chart doesn't use ClusterClass. I don't know if that's the explicit intention, it's more likely that we got that chart working with ASO and haven't expanded it (or cloned it) to support ClusterClass as well. Contributions are always welcome!

cc: @dtzar
